protocol/0014-ORDT-order_types.md

```diff
@@ -49,6 +49,117 @@ Notes on scope of current version of this spec:
+
+* `minimum peak size` must be `>` 0 and `≤ initial peak size`
+
+
+#### Execution
+
+* Glassberg orders trade just like non-glassberg persistent order, as if the order entered the book with quantity = initial peak size on submission and again each time they are refreshed.
+That is:
+
+    * On entry, unlike normal orders, `displayed quantity` is set to `initial peak size` not `quantity`.
+    
+    * As for any other order, `remaning == quantity` on entry.
+
+* The maximum size for a trade involving a glassberg order is the `displayed quantity` immediataly prior to the trade.
+(This is technically also true for a normal order, given that for non-glassberg orders `displayed quantity == remaining`.)
+
+* When a glassberg order trades, both `remaining` and `displayed quantity` are reduced by the trade size.
+
+* Glassberg orders can trade many times without refresh, reducing `displayed quantity` each time.
+The order will not be refreshed after each trade while `displayed quantity ≥ minimum peak size`.
+
+* The order will also not be refreshed within a block even if multiple trades occur or `displayed quantity == 0` if the refresh policy is set to `BLOCK_END`. 
+
+* Glassberg orders never trade more than their `displayed quantity` at the start of the transaction, as the result of any one transaction.
+
+* When `displayed quantity < minimum peak size` and `remaining > displayed quantity` the order will be refreshed:
+
+    * The refresh either happens at the end of the transaction when the order became eligible for refresh, if the `refresh policy == IMMEDIATE`; or at the end of the block containing that transaction if the `refresh policy == BLOCK_END`
+
+    * On refresh `display quantity` is set to `min(remaining, initial peak size)`.
+
+* Once the remaning quantity is equal to the displayed quantity, no further refresh is possible.
+The order now behaves like a normal limit order and will leave the book if it trades away completely.
+
+
+#### Amendment
+
+* Amending the size of a glassberg order amends the total remaining quantity and leaves the displayed quantiy unaffected unless the new remaining quanity is smaller than the current displayed quantity, in which case the displayed quantity is reduced to the total remaining quantity.
+
+
+#### Auctions
+
+* Glassbergs can be entered or carried into auctions if the underlying TIF is supported. 
+
+* Glassbergs can trade in the auction uncrossing up to their current size as for any other transaxction that would cause a trade.
+
+* Glassbergs are refreshed after an auction uncrossing if they traded away, according to the same rules for the refresh policy as for normal execution.
+
+
+#### APIs
+
+* The fields `displayed quantity`, `remaining`, `quantity`, `initial peak size`, `minimum peak size`, `refresh policy` must be exposed by data node APIs in addition to all normal fields for an order.
+
+* Glassberg refresh must generate an event bus event.
+
+
+
```

Comment:
I would add a comment that an LP that meets their obligation via glassbergs that refresh at end of block will always meet the SLA commitment (though perhaps that's obvious as we're counting time for SLA and it never moves inside a block...).

Reply:
Added this explicitly to [0042](https://github.com/vegaprotocol/specs/pull/1685/files#diff-71887ef730ec45323f5184892e551217f5e434a46bbabc0574a33a2a2d2d837aR230-R231)

Reply:
The above note will be removed if we do [this](https://github.com/vegaprotocol/specs/pull/1685#discussion_r1182395475)

Reply:
Amended, the note now reads as below:

>   * Note that this happens _before_ any glassberg order refreshes with `refresh policy == BLOCK_END`, even if `k == N`, and after glassbergs refreshed with `refresh policy == IMMEDIATELY` that need refreshing as a result of transaction `k`. This means that while a glassberg has sufficient `remaining` quantity, it will **never** be considered to be contributing less than its `initial peak size` of liquidity if it's policy is `IMMEDIATELY`. A `BLOCK_END` glassberg will **never** be refreshed within before the SLA achievement is evaluated for that block, and can therefore only be guaranteed to be contributing at least `minimum peak size` of liquidity, even when it has plenty of remaining volume.